src/main/java/com/bookshop/bazydanych/product/ProductService.java

```diff
@@ -75,10 +83,49 @@ public Product getbyId(Long id){
         return productRepository.getById(id);
     }
 
-    public void updateProduct(Product product){
-        Product productToUpdate = productRepository.getOne(product.getId());
-        productToUpdate = product;
-        productRepository.save(productToUpdate);
-        //TODO dodawanie produktów!
+    public void updateProduct(Product productOld){
+        Product product = new Product(
+                productOld.getName(),
+                productOld.getUnit(),
+                productOld.getProducent(),
+                productOld.getStock(),
+                productOld.getPrice(),
+                productOld.getStatus(),
+                productOld.getDescription(),
+                productOld.getCurrency(),
+                productOld.getCategory(),
+                productOld.getPlatform());
+
+        ArrayList<Long> orders = new ArrayList<>();
+        orderService.getAllOrders().forEach(a -> {
+            a.getProductIds().forEach(b ->{
+                if(b == product.getId()){
+                    if(orders.indexOf(a.getId())<0){
+                        orders.add(a.getId());
+                    }
+                }
+            });
+        });
+
+        if(orders.isEmpty()){
+            getActive().forEach(a -> {
```

Comment:
pobierasz wszystkie aktywne i wybierasz ten którego nazwa się zgadza? dude id sluży do indetyfikacji to raz dwa przecież wystarczy zrobić deactivateProduct(product.getId())